Library/Homebrew/extend/os/mac/keg.rb

```diff
@@ -31,36 +31,60 @@ def codesign_patched_binary(file)
     return unless Hardware::CPU.arm?
 
     odebug "Codesigning #{file}"
-    # Use quiet_system to squash notifications about resigning binaries
-    # which already have valid signatures.
-    return if quiet_system("codesign", "--sign", "-", "--force",
-                           "--preserve-metadata=entitlements,requirements,flags,runtime",
-                           file)
+    prepare_codesign_writable_files(file) do
+      # Use quiet_system to squash notifications about resigning binaries
+      # which already have valid signatures.
+      return if quiet_system("codesign", "--sign", "-", "--force",
+                             "--preserve-metadata=entitlements,requirements,flags,runtime",
+                             file)
 
-    # If the codesigning fails, it may be a bug in Apple's codesign utility
-    # A known workaround is to copy the file to another inode, then move it back
-    # erasing the previous file. Then sign again.
-    #
-    # TODO: remove this once the bug in Apple's codesign utility is fixed
-    Dir::Tmpname.create("workaround") do |tmppath|
-      FileUtils.cp file, tmppath
-      FileUtils.mv tmppath, file, force: true
+      # If the codesigning fails, it may be a bug in Apple's codesign utility
+      # A known workaround is to copy the file to another inode, then move it back
+      # erasing the previous file. Then sign again.
+      #
+      # TODO: remove this once the bug in Apple's codesign utility is fixed
+      Dir::Tmpname.create("workaround") do |tmppath|
+        FileUtils.cp file, tmppath
+        FileUtils.mv tmppath, file, force: true
+      end
+
+      # Try signing again
+      odebug "Codesigning (2nd try) #{file}"
+      result = system_command("codesign", args: [
+        "--sign", "-", "--force",
+        "--preserve-metadata=entitlements,requirements,flags,runtime",
+        file
+      ], print_stderr: false)
+      return if result.success?
+
+      # If it fails again, error out
+      onoe <<~EOS
+        Failed applying an ad-hoc signature to #{file}:
+        #{result.stderr}
+      EOS
     end
+  end
 
-    # Try signing again
-    odebug "Codesigning (2nd try) #{file}"
-    result = system_command("codesign", args: [
-      "--sign", "-", "--force",
-      "--preserve-metadata=entitlements,requirements,flags,runtime",
-      file
+  def prepare_codesign_writable_files(file)
+    tmppath = Tempfile.new("changed_files")
+    system_command("codesign", args: [
+      "--display", "--file-list", tmppath, file
```

Comment:
I wonder if we can avoid the temporary file if we dumped this to stdout by passing `-` to `--file-list` instead.

Reply:
Yes, this does seem to work since the actual output from `codesign` goes to `stderr`. I'll have a patch later today.